chalk-solve/src/recursive.rs

```diff
@@ -142,22 +142,9 @@ impl<'me, I: Interner> Solver<'me, I> {
                 // None of our subgoals depended on us directly.
                 // We can return.
                 self.context.search_graph[dfn].solution = current_answer;
-                self.context.search_graph[dfn].solution_priority = current_prio;
                 return *minimums;
             }
 
-            let old_answer = &self.context.search_graph[dfn].solution;
```

Comment:
so was all the priority handling here just not necessary, or is it somehow subsumed by the negative cycle handling? :thinking:

Reply:
I think this comment is referring to the changes in the first commit, which remove the logic that merges the "previous answer" (from prior iteration rounds) with the "new answer" (from the current iteration round). I don't believe that logic was *ever* necessary, precisely because of the monotonic nature of solving -- i.e., each round should be producing strictly more answers than the round before, so it should include whatever answers we were getting before. The only exception would be negative cycles, but we are always producing the same result for those ("ambiguous").

Reply:
Hmm I do remember adding it for a reason -- though I wonder whether I didn't manage to write a test for it, or whether that just works now. I think it was related to the fact that the priority logic does break this monotonic nature -- you can get a solution involving `Trait::AssocType<>` in the first round, and then get the properly normalized type in the second round. (The priority logic could maybe be interpreted as having a "not (the normalize rule applies)" in the fallback rule?)

Reply:
Hmm, good point! I remember us discussing this a bit. Sigh, things are complex. =) I guess we should see if we can craft an example where this actually causes a *problem*. I can imagine some cases where it might.

One situation would be around the infamous logic like

```rust
trait SomeTrait {
  type Item: Eq;
}

fn foo<T, X: SomeTrait<Item=T>>() {
}
```

where we (in rustc) fail to figure out that `T: Eq`.

Reply:
Still, I *think* it was important to remove that "merge with the old result" logic -- but maybe more for the coinductive case than for this negative cycle handling, actually.